api/v1alpha1/nodepool_types.go

```diff
@@ -55,152 +55,281 @@ type NodePool struct {
 	// Spec is the desired behavior of the NodePool.
 	Spec NodePoolSpec `json:"spec,omitempty"`
 
-	// Status is the most recently observed status of the NodePool.
+	// Status is the latest observed status of the NodePool.
 	Status NodePoolStatus `json:"status,omitempty"`
 }
 
-// NodePoolSpec defines the desired state of NodePool
+// NodePoolSpec is the desired behavior of a NodePool.
 type NodePoolSpec struct {
-	// ClusterName is the name of the Cluster this object belongs to.
+	// ClusterName is the name of the HostedCluster this NodePool belongs to.
+	//
+	// TODO(dan): Should this be a LocalObjectReference?
+	//
 	// +immutable
 	ClusterName string `json:"clusterName"`
+
+	// Release specifies the OCP release used for the NodePool. This informs the
+	// ignition configuration for machines, as well as other platform specific
+	// machine properties (e.g. an AMI on the AWS platform).
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Release Release `json:"release"`
+
+	// Platform specifies the underlying infrastructure provider for the NodePool
+	// and is used to configure platform specific behavior.
+	//
+	// +immutable
+	Platform NodePoolPlatform `json:"platform"`
+
+	// NodeCount is the desired number of nodes the pool should maintain. If
+	// unset, the default value is 0.
+	//
 	// +optional
 	NodeCount *int32 `json:"nodeCount"`
 
-	// +kubebuilder:validation:Optional
-	// TODO (alberto): this ConfigMaps are meant to contain
-	// MachineConfig, KubeletConfig and ContainerRuntimeConfig but
-	// MCO only supports MachineConfig in bootstrap mode atm
-	// https://github.com/openshift/machine-config-operator/blob/9c6c2bfd7ed498bfbc296d530d1839bd6a177b0b/pkg/controller/bootstrap/bootstrap.go#L104-L119
-	// By contractual convention the ConfigMap structure is as follow:
-	// type: ConfigMap
-	//   data:
-	//     config: |-
-	Config []v1.LocalObjectReference `json:"config,omitempty"`
-
-	Management NodePoolManagement `json:"nodePoolManagement"`
+	// Management specifies behavior for managing nodes in the pool, such as
+	// upgrade strategies and auto-repair behaviors.
+	Management NodePoolManagement `json:"management"`
```

Comment:
This JSON tag didn't match the field before, which I assume was just a mistake we missed. So this is technically a breaking change for any existing YAML floating around out there, but I feel like we should fix it now. Thoughts?

Reply:
Yes, it's a typo. We should definitely fix it.